cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -199,8 +199,9 @@ void addHessInPt(algorithmFPType * h, const algorithmFPType * xi, const algorith
 }
 
 template <typename algorithmFPType, Method method, CpuType cpu>
-services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::doCompute(const algorithmFPType * x, const algorithmFPType * y, size_t nRows,
-                                                                                 size_t n, size_t p, NumericTable * betaNT, NumericTable * valueNT,
+services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::doCompute(const NumericTable * dataNT,
```

Comment:
prefer reference over pointers when you don't need to check for null. Otherwise, should be checks for nullptr

```suggestion
services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::doCompute(const NumericTable & dataNT,
```

or 

```suggestion
services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::doCompute(const NumericTablePtr & dataNT,
```

Reply:
I decided use ptrs with checks